### PR TITLE
crimson/osd: drop repop if osdmap has increased

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Fixes: https://tracker.ceph.com/issues/47310
Signed-off-by: Xuehan Xu <xxhdx1985126@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by xxhdx1985126** [comment]
@tchaikov Please take a look, thanks:-)

**Comment by tchaikov** [comment]
let's discuss this next Monday.

**Comment by xxhdx1985126** [comment]
@tchaikov I have been able to move ClientRequest's relative code in to PG::handle_osd_op, because ClientRequest's processing involves multiple blocking_futures which can only be processed in ClientRequest. Please take a look, thanks:)

File: src/crimson/osd/pg.cc [hunk 1]
```diff
@@ -931,12 +935,18 @@ seastar::future<> PG::handle_rep_op(Ref<MOSDRepOp> req)
 void PG::handle_rep_op_reply(crimson::net::Connection* conn,
 			     const MOSDRepOpReply& m)
 {
-  if (is_valid_rep_op_reply(m)) {
+  if (!can_discard_replica_op(m)) {
```
Comment (tchaikov) [inline src/crimson/osd/pg.cc:938]: could you please extract the rename change into a separate commit ?

Comment (xxhdx1985126) [reply src/crimson/osd/pg.cc:938]: ok, that shall be done:-)

**Comment by xxhdx1985126** [comment]
@tchaikov Done:-)

**Review by tchaikov — APPROVED** [review]
